waspc/data/Generator/templates/sdk/wasp/core/auth.js

```diff
@@ -27,119 +26,16 @@ const auth = handleRejection(async (req, res, next) => {
-          include: {
-            identities: true
-          }
-        }
-      }
-    })
-  if (!user) {
+  if (!session || !user) {
     throwInvalidCredentialsError()
   }
 
-  // TODO: This logic must match the type in types/index.ts (if we remove the
-  // password field from the object here, we must to do the same there).
-  // Ideally, these two things would live in the same place:
-  // https://github.com/wasp-lang/wasp/issues/965
-  let sanitizedUser = { ...user }
-  sanitizedUser.auth.identities = sanitizedUser.auth.identities.map(identity => {
-    identity.providerData = deserializeAndSanitizeProviderData(identity.providerData, { shouldRemovePasswordField: true })
-    return identity
-  });
-  return sanitizedUser
-}
-
-const SP = new SecurePassword()
-
-export const hashPassword = async (password) => {
-  const hashedPwdBuffer = await SP.hash(Buffer.from(password))
-  return hashedPwdBuffer.toString("base64")
-}
-
-export const verifyPassword = async (hashedPassword, password) => {
-  const result = await SP.verify(Buffer.from(password), Buffer.from(hashedPassword, "base64"))
-  if (result !== SecurePassword.VALID) {
-    throw new Error('Invalid password.')
-  }
-}
+  req.sessionId = session.id
+  req.user = user
 
-// Generates an unused username that looks similar to "quick-purple-sheep-91231". 
-// It generates several options and ensures it picks one that is not currently in use.
-export function generateAvailableDictionaryUsername() {
-  const adjectives = ['fuzzy', 'tall', 'short', 'nice', 'happy', 'quick', 'slow', 'good', 'new', 'old', 'first', 'last', 'old', 'young']
-  const colors = ['red', 'green', 'blue', 'white', 'black', 'brown', 'purple', 'orange', 'yellow']
-  const nouns = ['wasp', 'cat', 'dog', 'lion', 'rabbit', 'duck', 'pig', 'bee', 'goat', 'crab', 'fish', 'chicken', 'horse', 'llama', 'camel', 'sheep']
-
-  const potentialUsernames = []
-  for (let i = 0; i < 10; i++) {
-    const potentialUsername = `${adjectives[randomInt(adjectives.length)]}-${colors[randomInt(colors.length)]}-${nouns[randomInt(nouns.length)]}-${randomInt(100_000)}`
-    potentialUsernames.push(potentialUsername)
-  }
-
-  return findAvailableUsername(potentialUsernames)
-}
-
-// Generates an unused username based on an array of username segments and a separator. 
-// It generates several options and ensures it picks one that is not currently in use.
-export function generateAvailableUsername(usernameSegments, config) {
-  const separator = config?.separator || '-'
-  const baseUsername = usernameSegments.join(separator)
-
-  const potentialUsernames = []
-  for (let i = 0; i < 10; i++) {
-    const potentialUsername = `${baseUsername}${separator}${randomInt(100_000)}`
-    potentialUsernames.push(potentialUsername)
-  }
-
-  return findAvailableUsername(potentialUsernames)
-}
-
-// Checks the database for an unused username from an array provided and returns first.
-async function findAvailableUsername(potentialUsernames) {
```

Comment:
Where did these two guys end up?

Reply:
We removed them since they are no longer useful / used. They are not mentioned in the docs either.

We used them before to ensure that there is some sort of a default username for Google login when using username + Google auth combo. Now, with the latest auth schema changes, this is no longer needed since username and Google have their own separate data.